lib/ocpp/v16/smart_charging.cpp

```diff
@@ -375,16 +375,20 @@ void SmartChargingHandler::clear_all_profiles() {
     }
 }
 
-std::vector<ChargingProfile> SmartChargingHandler::get_valid_profiles(const ocpp::DateTime& start_time,
-                                                                      const ocpp::DateTime& end_time,
-                                                                      const int connector_id) {
+std::vector<ChargingProfile>
+SmartChargingHandler::get_valid_profiles(const ocpp::DateTime& start_time, const ocpp::DateTime& end_time,
+                                         const int connector_id,
+                                         const std::set<ChargingProfilePurposeType>& purposes_to_ignore) {
     std::vector<ChargingProfile> valid_profiles;
 
     {
         std::lock_guard<std::mutex> lk(charge_point_max_profiles_map_mutex);
 
-        for (const auto& [stack_level, profile] : stack_level_charge_point_max_profiles_map) {
-            valid_profiles.push_back(profile);
+        if (std::find(std::begin(purposes_to_ignore), std::end(purposes_to_ignore),
```

Comment:
I would be tempted to use std::any_of since access to the found element isn't needed.